javagen/src/main/java/com/azure/autorest/mapper/ModelPropertyMapper.java

```diff
@@ -129,7 +136,7 @@ public ClientModelProperty map(Property property) {
         List<String> annotationArgumentList = new ArrayList<String>() {{
             add(String.format("value = \"%s\"", xmlParamName));
         }};
-        if (property.isRequired()) {
+        if (property.isRequired() && !propertyIsSecret) {
             annotationArgumentList.add("required = true");
         }
```

Comment:
Code happens here.

Note it apply to both Typespec and Swagger (from m4). Let me know if this be a concern for Swagger.